app/src/main/java/com/pvp/app/ui/screen/settings/SettingsScreen.kt

```diff
@@ -207,61 +247,98 @@ fun SettingCard(
-            verticalAlignment = Alignment.CenterVertically
-        ) {
-            Text(
-                color = MaterialTheme.colorScheme.onPrimary,
-                style = MaterialTheme.typography.bodyMedium,
-                text = value
-            )
+        if (value is Boolean) {
+            Row(
+                modifier = Modifier.align(Alignment.End),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Switch(
+                    checked = value,
+                    onCheckedChange = {
+                        onEdit()
+                    },
+                    thumbContent = if (value) {
+                        {
+                            Icon(
+                                imageVector = Icons.Filled.Check,
+                                contentDescription = null,
+                                modifier = Modifier.size(SwitchDefaults.IconSize),
+                            )
+                        }
+                    } else {
+                        null
+                    },
+                    colors = SwitchDefaults.colors(
+                        checkedTrackColor = MaterialTheme.colorScheme.surface,
+                        uncheckedTrackColor = MaterialTheme.colorScheme.surfaceDim,
+                        checkedThumbColor = MaterialTheme.colorScheme.outline,
+                        uncheckedThumbColor = MaterialTheme.colorScheme.outline,
+                        checkedBorderColor = MaterialTheme.colorScheme.outline,
+                        uncheckedBorderColor = MaterialTheme.colorScheme.outline
+                    ),
+                    enabled = isEnabled
+                )
+            }
+        } else if (value != null) {
+            var dialogOpen by remember { mutableStateOf(false) }
 
-            Spacer(modifier = Modifier.size(8.dp))
+            Row(
+                modifier = Modifier
+                    .align(Alignment.End)
+                    .clickable {
+                        if (isEnabled) {
+                            dialogOpen = true
+                        }
+                    },
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Text(
+                    color = textColor,
+                    style = MaterialTheme.typography.bodyMedium,
+                    text = value.toString()
+                )
 
-            Icon(
-                tint = MaterialTheme.colorScheme.onPrimary,
-                contentDescription = "Open dialog to edit a setting",
-                imageVector = Icons.Outlined.Edit
-            )
-        }
+                Spacer(modifier = Modifier.size(8.dp))
 
-        if (state) {
-            Dialog(
-                onDismissRequest = {
-                    onEdit()
+                Icon(
+                    tint = textColor,
+                    contentDescription = "Open dialog to edit a setting",
```

Comment:
This should refer to which setting more as all the other accessibility support properties. In case user is blind and is using sound device's help to navigate, each setting icon will just say "Open a dialog to edit a setting". But what setting? And is this icon or something else?

If we are providing accessibility support, let's make it half decent or not provide it at all.